server.go

```diff
@@ -486,49 +499,83 @@ func NewFromConfig(config ServerConfig) (*Server, error) {
 	// found during setup we should use logger.Fatal or
 	// logger.Panic, because that will give us breakage monitoring
 	// through Sentry.
-	numWorkers := 1
-	if conf.NumWorkers > 1 {
-		numWorkers = conf.NumWorkers
-	}
-	logger.WithField("number", numWorkers).Info("Preparing workers")
-	// Allocate the slice, we'll fill it with workers later.
-	ret.Workers = make([]*Worker, numWorkers)
+
 	ret.numReaders = conf.NumReaders
 
 	// This must come before worker initialization. We need to
 	// initialize workers with state from *Server.IsWorker.
 	ret.ForwardAddr = conf.ForwardAddress
 
-	// Control whether Veneur should emit metric
-	// "veneur.flush.unique_timeseries_total", which may come at a
-	// slight performance hit to workers.
-	ret.CountUniqueTimeseries = conf.CountUniqueTimeseries
+	ret.enableMetricRouting = conf.Features.EnableMetricRouting
+	ret.computationRoutingConfig = conf.MetricComputationRouting
+	ret.WorkerSets = make([]WorkerSet, 0)
+	ret.subscribedFlushGroupsBySink = make(map[string][]string)
+	if ret.enableMetricRouting {
+		for _, config := range ret.computationRoutingConfig {
+			if config.WorkerCount < 1 {
+				logger.WithFields(logrus.Fields{
+					"worker_name":  config.Name,
+					"worker_count": config.WorkerCount,
+				}).Fatal("WorkerCount must be greater than 0")
+			}
+			logger.WithField("worker_name", config.Name).Info("Preparing WorkerSet")
+			workerSet := WorkerSet{make([]*Worker, config.WorkerCount), &config}
+			for i := 0; i < config.WorkerCount; i++ {
+				// TODO: ids are not unique like this?
```

Comment:
ids are only used for logging purposes here, but I still need to clean this up - it seems like maybe something useful here would be "WorkerSet name + some UUID"